### PR TITLE
codebase/generate-values-for-entity-attributes-in-hibernate [BAEL-8327]

### PR DESCRIPTION
**Pull request by hardikSinghBehl**
Codebase for the article: **Generate Values for Entity Attributes in Hibernate**

### CODE REVIEW

File: persistence-modules/hibernate6/src/test/java/com/baeldung/attributevaluegenerator/WizardAttributeValueGeneratorIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+    @Test
+    void whenNewWizardRecordSaved_thenHouseGetsSorted() {
+        Wizard wizard = new Wizard();
+        wizard.setId(UUID.randomUUID());
+        wizard.setName(RandomString.make());
+
+        Wizard savedWizard = wizardRepository.save(wizard);
+
+        assertThat(savedWizard.getHouse()).isNotBlank();
```
Comment (theangrydev) [inline persistence-modules/hibernate6/src/test/java/com/baeldung/attributevaluegenerator/WizardAttributeValueGeneratorIntegrationTest.java:28]: Let's indent even these one liners too for consistency
```suggestion
        assertThat(savedWizard.getHouse())
            .isNotBlank();
```

File: persistence-modules/hibernate6/src/test/java/com/baeldung/attributevaluegenerator/WizardAttributeValueGeneratorIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        wizard.setId(UUID.randomUUID());
+        wizard.setName(RandomString.make());
+
+        Wizard savedWizard = wizardRepository.save(wizard);
+        LocalDateTime initialUpdatedAtTimestamp = savedWizard.getUpdatedAt();
+
+        savedWizard.setName(RandomString.make());
+        Wizard updatedWizard = wizardRepository.save(savedWizard);
+
+        assertThat(updatedWizard.getUpdatedAt()).isAfter(initialUpdatedAtTimestamp);
```
Comment (theangrydev) [inline persistence-modules/hibernate6/src/test/java/com/baeldung/attributevaluegenerator/WizardAttributeValueGeneratorIntegrationTest.java:43]: Let's indent even these one liners too for consistency
```suggestion
        assertThat(updatedWizard.getUpdatedAt())
            .isAfter(initialUpdatedAtTimestamp);
```

File: persistence-modules/hibernate6/src/test/java/com/baeldung/attributevaluegenerator/WizardAttributeValueGeneratorIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    void whenNewWizardRecordSaved_thenSpellPowerGenerated() {
+        Wizard wizard = new Wizard();
+        wizard.setId(UUID.randomUUID());
+        wizard.setName(RandomString.make());
+
+        Wizard savedWizard = wizardRepository.save(wizard);
+
+        assertThat(savedWizard.getSpellPower())
+          .isNotNull()
+          .isGreaterThan(50);
```
Comment (theangrydev) [inline persistence-modules/hibernate6/src/test/java/com/baeldung/attributevaluegenerator/WizardAttributeValueGeneratorIntegrationTest.java:56]: In the PR we still use a 4 space indent, it's only 2-space in the article
```suggestion
        assertThat(savedWizard.getSpellPower())
            .isNotNull()
            .isGreaterThan(50);
```

You can use the automatic Baeldung formatter if you import the IDE formatter profile

File: persistence-modules/hibernate6/src/main/java/com/baeldung/attributevaluegenerator/SortingHatHouseGenerator.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+import java.util.Random;
+
+import org.hibernate.engine.spi.SharedSessionContractImplementor;
+import org.hibernate.generator.BeforeExecutionGenerator;
+import org.hibernate.generator.EventType;
+
+public class SortingHatHouseGenerator implements BeforeExecutionGenerator {
+
+    private static final String[] HOUSES = { "Gryffindor", "Hufflepuff", "Ravenclaw", "Slytherin" };
+    private static final Random RANDOM = new Random();
```
Comment (theangrydev) [inline persistence-modules/hibernate6/src/main/java/com/baeldung/attributevaluegenerator/SortingHatHouseGenerator.java:13]: It doesn't matter that much but you should generally never use Random any more, always use ThreadLocalRandom
```suggestion
    private static final Random RANDOM = new ThreadLocalRandom();
```

**Review by theangrydev — CHANGES_REQUESTED** [review]

File: persistence-modules/hibernate6/src/main/java/com/baeldung/attributevaluegenerator/SortingHatHouseGenerator.java (hunk 4 above)
Comment (hardikSinghBehl) [reply persistence-modules/hibernate6/src/main/java/com/baeldung/attributevaluegenerator/SortingHatHouseGenerator.java]: got it, thanks. Incorporated.

File: persistence-modules/hibernate6/src/test/java/com/baeldung/attributevaluegenerator/WizardAttributeValueGeneratorIntegrationTest.java (hunk 3 above)
Comment (hardikSinghBehl) [reply persistence-modules/hibernate6/src/test/java/com/baeldung/attributevaluegenerator/WizardAttributeValueGeneratorIntegrationTest.java]: review incorporated.

File: persistence-modules/hibernate6/src/test/java/com/baeldung/attributevaluegenerator/WizardAttributeValueGeneratorIntegrationTest.java (hunk 1 above)
Comment (hardikSinghBehl) [reply persistence-modules/hibernate6/src/test/java/com/baeldung/attributevaluegenerator/WizardAttributeValueGeneratorIntegrationTest.java]: review incorporated.

**Review by theangrydev — APPROVED** [review]

**Pull request merged**